core/src/main/java/io/questdb/cairo/map/FastMap.java

```diff
@@ -480,22 +482,21 @@ public void putRecord(Record value) {
             // no-op
         }
 
-        private MapValue createValue(FastMapValue value) {
-            int keySize = commit();
-            // calculate hash remembering "key" structure
-            // [ key size | key block | value block ]
-            int hashCode = hash();
+        private MapValue createValue(int keySize, int hashCode) {
             int index = hashCode & mask;
             long offset = getOffset(offsets, index);
+            if (offset > -1 && hashCode == getHashCode(offsets, index) && eq(offset)) {
```

Comment:
nit: I think this can be better jitted the cases where new value is created more often an existing found, which is hot path in case of shard merges, perhaps on shard merge this variant is better to be called:

```
        private MapValue createValue(int keySize, int hashCode) {
            int index = hashCode & mask;
            long offset = getOffset(offsets, index);
            if (offset < 0) {
                return asNew(this, index, hashCode, value);
            }
            return createValueSlow0(value, offset, index, hashCode, keySize);
        }

```

Reply:
Changed the order and simplified in 4051076c1e54ac60f26f5c7fe18e78fc2fb9a2bb